Updates for Aggregating-,Collapsing-, Replacing- and SummingMergeTree.

BayoNet opened the pull request:
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

BayoNet committed:
Update of english version of descriprion of the table function `file`.

BayoNet committed:
New syntax for ReplacingMergeTree.
Some improvements in text.

BayoNet committed:
Significantly change article about SummingMergeTree.
Article is restructured, text is changed in many places of the document. New syntax for table creation is described.

BayoNet committed:
Descriptions of AggregateFunction and AggregatingMergeTree are updated. Russian version.

BayoNet committed:
New syntax for new syntax of CREATE TABLE

BayoNet committed:
Added english docs on Aggregating, Replacing and SummingMergeTree.

BayoNet committed:
CollapsingMergeTree docs. English version.

BayoNet committed:
1. Update of CollapsingMergeTree. 2. Minor changes in markup

BayoNet committed:
Merge branch 'updating-docs'

alexey-milovidov committed:
Update aggregatefunction.md

alexey-milovidov committed:
Update aggregatefunction.md

alexey-milovidov commented on docs/en/data_types/nested_data_structures/aggregatefunction.md:
```diff
@@ -1,4 +1,64 @@
+
+In contrast to the corresponding functions `uniq` and `quantiles`, `-State`- functions return the state, instead the final value. In other words, they return a value of `AggregateFunction` type.
+
+In the results of `SELECT` query the values of  `AggregateFunction` type have implementation-specific binary representation for all of the ClickHouse output formats. If dump data into, for example, `TabSeparated`  format with `SELECT`  query  then this dump can be loaded back using `INSERT` query.
```
Why so many double whitespaces?

alexey-milovidov committed:
Update aggregatefunction.md

alexey-milovidov committed:
Update aggregatingmergetree.md

alexey-milovidov commented on docs/en/operations/table_engines/aggregatingmergetree.md:
````diff
@@ -1,59 +1,66 @@
-```sql
-CREATE TABLE t
+```
+CREATE [TEMPORARY] TABLE [IF NOT EXISTS] [db.]name [ON CLUSTER cluster]
````
Temporary table cannot have engine.

alexey-milovidov reviewed (commented):
.

BayoNet committed:
Merge remote-tracking branch 'upstream/master'

BayoNet committed:
GraphiteMergeTree docs update.
New syntax for creation of Replicated* tables.
Minor changes in *MergeTree tables creation syntax.

BayoNet committed:
Markup fix

blinkov commented on docs/en/operations/table_engines/aggregatingmergetree.md:
````diff
@@ -1,59 +1,66 @@
-
-``` sql
-SELECT uniq(UserID) FROM table
+!!! attention    Do not use this method in new projects and, if possible, switch the old projects to the method described above.
````
lost newline after attention

BayoNet replied:
fixed

blinkov commented on docs/en/operations/table_engines/aggregatingmergetree.md:
```diff
@@ -1,59 +1,66 @@
+**Query clauses**
 
-In contrast to the corresponding `uniq` and `quantiles` functions, these functions return the state, rather than the prepared value. In other words, they return an `AggregateFunction` type value.
+When creating a `ReplacingMergeTree` table the same [clauses](mergetree.md#table_engines-mergetree-configuring)  are required, as when creating a `MergeTree`  table.
```
double spaces

BayoNet replied:
fixed

blinkov commented on docs/en/operations/table_engines/aggregatingmergetree.md:
```diff
@@ -1,59 +1,66 @@
+## SELECT and INSERT
 
-With SELECT from an `AggregatingMergeTree` table, use GROUP BY and aggregate functions with the '-Merge' modifier in order to complete data aggregation.
+To insert data, use `INSERT SELECT` query with aggregate `- State`- functions.
```
link to explanation of what INSERT SELECT is?

blinkov commented on docs/en/operations/table_engines/aggregatingmergetree.md:
```diff
@@ -1,59 +1,66 @@
+When selecting data from `AggregatingMergeTree`  table, use `GROUP BY` clause and the same aggregate functions as when inserting data, but using `- Merge` suffix.
 
-Example:
+In the results of `SELECT` query the values of  `AggregateFunction` type have implementation-specific binary representation for all of the ClickHouse output formats. If dump data into, for example, `TabSeparated`  format with `SELECT`  query  then this dump can be loaded back using `INSERT` query.
```
more double spaces

blinkov commented on docs/en/operations/table_engines/collapsingmergetree.md:
```diff
@@ -2,39 +2,218 @@
 # CollapsingMergeTree
 
-*This engine is used specifically for Yandex.Metrica.*
+The engine inherits from [MergeTree](mergetree.md#table_engines-mergetree) and adds to data parts merge algorithm the logic of rows collapsing.
```
... adds the logic of ... to ... algorithm.

blinkov commented on docs/en/operations/table_engines/collapsingmergetree.md:
```diff
@@ -2,39 +2,218 @@
+The engine inherits from [MergeTree](mergetree.md#table_engines-mergetree) and adds to data parts merge algorithm the logic of rows collapsing.
 
-It differs from `MergeTree` in that it allows automatic deletion, or "collapsing" certain pairs of rows when merging.
+`CollapsingMergeTree` deletes (collapses) pairs of rows by specific rules. The engine may significantly reduce the volume of storage and increase efficiency of `SELECT` query as a consequence.
```
1. asynchronously deletes
2. it makes sense to actually briefly explain the rules and explicitly say that rows without a pair are kept, before going into details

blinkov commented on docs/en/operations/table_engines/collapsingmergetree.md:
```diff
@@ -2,39 +2,218 @@
+
+**CollapsingMergeTree Parameters**
+
+- `sign` — Name of the column with the type of row: `1` — "state" row, `-1` — "cancel" row.
```
too many `—` signs in one row, better make it a list or replace with "means" (or synonym)

blinkov commented on docs/en/operations/table_engines/collapsingmergetree.md:
```diff
@@ -2,39 +2,218 @@
 
-1. Write a query with GROUP BY and aggregate functions that accounts for the sign. For example, to calculate quantity, write 'sum(Sign)' instead of 'count()'. To calculate the sum of something, write 'sum(Sign * x)' instead of 'sum(x)', and so on, and also add 'HAVING sum(Sign) `>` 0'. Not all amounts can be calculated this way. For example, the aggregate functions 'min' and 'max' can't be rewrit ...
-2. If you must extract data without aggregation (for example, to check whether rows are present whose newest values match certain conditions), you can use the FINAL modifier for the FROM clause. This approach is significantly less efficient.
+Getting the data
```
`:`

blinkov commented on docs/en/operations/table_engines/graphitemergetree.md:
```diff
@@ -4,56 +4,118 @@
-- `Time` – The time for measuring the metric.
-- `Value` – The value of the metric at the time set in Time.
-- `Version` – Determines which value of the metric with the same Path and Time will remain in the database.
+### Deprecated Method for Creating a Table
```
put this into `<details>`

blinkov commented on docs/en/operations/table_engines/aggregatingmergetree.md:
```diff
@@ -1,59 +1,66 @@
+When creating a `ReplacingMergeTree` table the same [clauses](mergetree.md#table_engines-mergetree-configuring)  are required, as when creating a `MergeTree`  table.
 
-An `AggregateFunction` type value can't be output in Pretty formats. In other formats, these types of values are output as implementation-specific binary data. The `AggregateFunction` type values are not intended for output or saving in a dump.
+### Deprecated Method for Creating a Table
```
put this into `<details>`

blinkov commented on docs/en/operations/table_engines/mergetree.md:
```diff
@@ -25,29 +24,47 @@ Main features:
     If necessary, you can set the data sampling method in the table.
 
-## Engine Configuration When Creating a Table
+!!!info
```
space

BayoNet committed:
Markup and language fixes

BayoNet committed:
Clarification in the CollapsingMergeTree article

blinkov commented:
@sundy-li @zhang2014 this PR touches few articles that are already have Chinese versions

zhang2014 commented:
Ok. I'll do the task : )
